core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProfileManagerDAO.scala

```diff
@@ -1,30 +1,79 @@
 package org.broadinstitute.dsde.rawls.billing
 
+import bio.terra.profile.model.{AzureManagedAppModel, CloudPlatform, CreateProfileRequest, ProfileModel}
 import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.RawlsExceptionWithErrorReport
 import org.broadinstitute.dsde.rawls.config.{AzureConfig, MultiCloudWorkspaceConfig}
 import org.broadinstitute.dsde.rawls.dataaccess.SamDAO
-import org.broadinstitute.dsde.rawls.model.{CreationStatuses, RawlsBillingProject, RawlsBillingProjectName, RawlsBillingProjectResponse, SamResourceAction, SamResourceTypeNames, SamUserResource, UserInfo, AzureManagedAppCoordinates}
+import org.broadinstitute.dsde.rawls.model.{AzureManagedAppCoordinates, CreationStatuses, ErrorReport, RawlsBillingAccountName, RawlsBillingProject, RawlsBillingProjectName, SamResourceAction, SamResourceTypeNames, SamUserResource, UserInfo}
 
+import java.util.UUID
 import scala.concurrent.{ExecutionContext, Future}
+import scala.jdk.CollectionConverters._
 
+/**
+ * Common interface for Billing Profile Manager operations
+ */
 trait BillingProfileManagerDAO {
-  def listBillingProfiles(userInfo: UserInfo, samUserResources: Seq[SamUserResource])(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]]
+  def createBillingProfile(displayName: String, billingInfo: Either[RawlsBillingAccountName, AzureManagedAppCoordinates], userInfo: UserInfo): Future[ProfileModel]
+
+  def listBillingProfiles(samUserResources: Seq[SamUserResource], userInfo: UserInfo)(implicit ec: ExecutionContext): Future[Seq[RawlsBillingProject]]
+
+  def listManagedApps(subscriptionId: UUID, userInfo: UserInfo): Future[Seq[AzureManagedAppModel]]
 }
 
 
+class ManagedAppNotFoundException(errorReport: ErrorReport) extends RawlsExceptionWithErrorReport(errorReport)
+
 /**
  * Facade over the billing profile manager service. This service will eventually by the source of truth for
  * billing profiles in the Terra system. For now, we are using this to layer in "external" billing profiles
  * for the purposes of testing Azure workspaces.
  */
-class BillingProfileManagerDAOImpl(samDAO: SamDAO, config: MultiCloudWorkspaceConfig) extends BillingProfileManagerDAO with LazyLogging {
+class BillingProfileManagerDAOImpl(samDAO: SamDAO,
+                                   apiClientProvider: BillingProfileManagerClientProvider,
+                                   config: MultiCloudWorkspaceConfig) extends BillingProfileManagerDAO with LazyLogging {
+
+
+  override def listManagedApps(subscriptionId: UUID, userInfo: UserInfo): Future[Seq[AzureManagedAppModel]] = {
```

Comment:
I would be tempted to make this method `listAzureManagedApps` for clarity, though perhaps the return type is sufficient for providing that contextual clue.

Reply:
Actually, this is interesting because someday (I believe), the BillingProfileManager will also support GCP, and yet GCP will never have managed apps. That increases my discomfort with this method.

Reply:
I wonder if it would be clearer if this method took `AzureManagedAppCoordinates` as an argument instead of `subscriptionId`. I get that it only needs the `subscriptionId`, but as-is the caller has to determine where to get the `subscriptionId` from.

Reply:
This method is intended to serve two purposes:
1. For the context of this PR, enumerate the managed apps and ensure one exists prior to going ahead with profile creation. BPM does this as I look at the create [profile code there](https://github.com/DataBiosphere/terra-billing-profile-manager/blob/54ae94586a2c1d7aed7166114a9ad4f83475d09e/service/src/main/java/bio/terra/profile/service/profile/flight/create/CreateProfileVerifyDeployedApplicationStep.java#L24), so I think we may want to forgo the call to this method altogether [here](https://github.com/broadinstitute/rawls/blob/6e84bd7ac99efe20bd84fdb5ae88c7ae3c97c134/core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectCreator.scala#L27)

_but_

2. We also need this method to populate the future managed apps dropdown in the UI. The user will provide a subscription ID and we will need to call this method to enumerate the managed apps in the subscription. At that time in the profile creation lifecycle, we will not have the tenant ID nor the managed resource group ID so I think signature needs to remain as-is.

Reply:
OK, makes sense about the signature.

If you don't call this method in the validate step, then it seems that you are at risk of finding out later that the managed app actually doesn't exist, requiring the execution of the rollback code. If that's true, I would stick with doing this in validate.